### PR TITLE
xmake: init at 2.6.7

### PR DESCRIPTION
**Pull request by hyqhyq3**
###### Description of changes

    description = "A cross-platform build utility based on Lua";
    homepage    = "https://xmake.io/";
    license     = licenses.asl20;

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/build-managers/xmake/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ lib, stdenv, fetchgit
+# optional list of extra waf tools, e.g. `[ "doxygen" "pytest" ]`
+
+}:
+let
+in
```
Comment (ajs124) [inline pkgs/development/tools/build-managers/xmake/default.nix:6]:
```suggestion
}:
```

this empty let block doesn't do anything

File: pkgs/development/tools/build-managers/xmake/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchgit {
+    url = "https://github.com/xmake-io/xmake";
+    rev = "v${version}";
+    sha256 = "sha256-e6V/RM07T6TCCxdIL47M9P7KiNA6v+mMOcSBZkSp3YQ=";
+  };
+
+  enableParallelBuilding = true;
+
+  installPhase = ''
+    make install prefix=$out
```
Comment (ajs124) [inline pkgs/development/tools/build-managers/xmake/default.nix:20]: Can't this use `installFlags = [ "prefix=${placeholder "out"}" ];`?

That way, the installPhase is untouched and preInstall and postInstall still get executed

File: pkgs/development/tools/build-managers/xmake/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{ lib, stdenv, fetchgit
+# optional list of extra waf tools, e.g. `[ "doxygen" "pytest" ]`
+
+}:
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/xmake/default.nix:4]:
```suggestion
{ lib, stdenv, fetchgit }:
```

File: pkgs/development/tools/build-managers/xmake/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+let
+in
+stdenv.mkDerivation rec {
+  pname = "xmake";
+  version = "2.6.7";
+
+  src = fetchgit {
+    url = "https://github.com/xmake-io/xmake";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/xmake/default.nix:12]: Please use fetchFromGitHub

Comment (hyqhyq3) [reply pkgs/development/tools/build-managers/xmake/default.nix:9]: It depends on git submodules, fetchFromGitHub didn't checkout recusive. that's leads to a compile error.

Comment (ShadowRZ) [reply pkgs/development/tools/build-managers/xmake/default.nix:9]: > It depends on git submodules, fetchFromGitHub didn't checkout recusive. that's leads to a compile error.

It's possible to fetch submodules with `fetchFromGitHub`, add `fetchSubmodules = true`.

**Comment by ShadowRZ** [comment]
BTW, you should rebase your new changes and force push again, [according to the guidelines](https://nixos.org/manual/nixpkgs/unstable/#submitting-changes-hotfixing-pull-requests).

**Pull request closed without merging**